app/src/main/java/com/skyous/trakkit/ui/navigation/core/MainNavigator.kt

```diff
@@ -0,0 +1,227 @@
+package com.skyous.trakkit.ui.navigation.core
+
+import android.os.Bundle
+import android.support.v4.app.Fragment
+import android.support.v4.app.FragmentActivity
+import android.support.v4.app.FragmentManager
+import android.util.Log
+import com.skyous.trakkit.R
+import com.skyous.trakkit.ui.ExploreFragment
+import com.skyous.trakkit.ui.HomeFragment
+import com.skyous.trakkit.ui.LibraryFragment
+import com.skyous.trakkit.ui.ProfileFragment
+import java.lang.ref.WeakReference
+
+/**
+ * Navigator for the bottom navigation
+ *
+ * @author Peter Abraham
+ * @since 10/18/2018
+ *
+ */
+
+class MainNavigator private constructor() {
```

Comment:
Better implementation of singletons: https://medium.com/@adinugroho/singleton-in-kotlin-better-approach-8c5e28a140a5